src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets

```diff
@@ -292,5 +296,26 @@
   
   <!-- This target collects all the resolved references that are used to actually compile. -->
   <Target Name="CollectResolvedCompilationReferencesDesignTime" DependsOnTargets="CompileDesignTime" Returns="@(ReferencePathWithRefAssemblies)" />
+  
+  <!-- This target collects all the things built by the project for the up to date check. -->
+  <!-- See CopyFileToOutputDirectory target -->
+  <Target Name="CollectBuiltDesignTime" DependsOnTargets="CompileDesignTime" Returns="@(UpToDateCheckBuilt)">
+    <ItemGroup>
+      <!-- Assembly output, bin and obj -->
+      <UpToDateCheckBuilt Condition="'$(CopyBuildOutputToOutputDirectory)' != 'false' and '$(SkipCopyBuildProduct)' != 'true'" Include="$(TargetPath)"/>
+      <UpToDateCheckBuilt Include="@(IntermediateAssembly)"/>
+
+      <!-- Documentation file, bin and obj -->
+      <UpToDateCheckBuilt Condition="'$(_DocumentationFileProduced)'=='true'" Include="@(FinalDocFile)"/>
+      <UpToDateCheckBuilt Condition="'$(_DocumentationFileProduced)'=='true'" Include="@(DocFileItem)"/>
+
+      <!-- Symbols, bin and obj -->
+      <UpToDateCheckBuilt Condition="'$(_DebugSymbolsProduced)'=='true'" Include="@(_DebugSymbolsIntermediatePath)"/>
+      <UpToDateCheckBuilt Condition="'$(_DebugSymbolsProduced)'=='true' and '$(SkipCopyingSymbolsToOutputDirectory)' != 'true' and '$(CopyOutputSymbolsToOutputDirectory)' != 'false'" Include="@(_DebugSymbolsOutputPath)"/>
+      
+      <!-- app.config -->
+      <UpToDateCheckBuilt Condition=" '@(AppConfigWithTargetPath)' != '' " Include="@(AppConfigWithTargetPath->'$(OutDir)%(TargetPath)')" Original="@(AppConfigWithTargetPath)"/>
```

Comment:
Is this always re-generated if the user just has one, and doesn't have the auto-generate option enabled?  Maybe it works because the source app.config is renamed?

Reply:
Not sure I totally understand the question. Basically, this is saying "if Console37.exe.config is older than app.config then the project is not up to date". The "Original" property means "compare this output to this particular input, not all the other sources". If that seems confusing, I can make it clearer by splitting it out to another item type.

---
In reply to: [161834077](https://github.com/dotnet/project-system/pull/3142#discussion_r161834077) [](ancestors = 161834077)

Reply:
app.config files happen in a few ways:

1. Project doesn't have an input app.config, option to generate binding redirects enabled - one will be generated as part of the build.
2. Project already has an app.config, option to generate binding redirects enabled - the app.config in the project is updated with binding redirects and copied to the output directory with the app name.
3. Project already has an app.config, option to generate binding redirects *disabled* - the app.config in the project is copied to the output directory with the app name.

This question was to ensure that case 3 works properly here.

Reply:
Yup, I did test case 3. In case #1, AppConfigWithTargetPath will be null, so we won't check anything in up to date. In case #2, the target app.config could be out of date even if app.config has been touched, but that should only happen when the references have been changed which is going to cause a build anyway. And in #3, we check the source against the target.